### PR TITLE
fix regression in the use of name in ProjectionPushdown

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Resolves https://github.com/apache/arrow-datafusion/issues/8223

## Rationale for this change

We found a bug while upgrading IOx to use the latest datafusion

Some of our queries began failing like this:

> "Optimizer rule 'push_down_projection' failed"

The regression was introduced in https://github.com/apache/arrow-datafusion/pull/7981 and is related to columns with periods in their names

## What changes are included in this PR?

1. Back out an unecessary change from https://github.com/apache/arrow-datafusion/pull/7981
2. Add test coverage

## Are these changes tested?
yes

## Are there any user-facing changes?
bug fix

### CODE REVIEW

**Comment by alamb** [comment]
FYI @gruuya

File: datafusion/optimizer/src/push_down_projection.rs [hunk 1]
```diff
@@ -228,7 +228,7 @@ impl OptimizerRule for PushDownProjection {
                 // Gather all columns needed for expressions in this Aggregate
                 let mut new_aggr_expr = vec![];
                 for e in agg.aggr_expr.iter() {
-                    let column = Column::from(e.display_name()?);
+                    let column = Column::from_name(e.display_name()?);
```
Comment (alamb) [inline datafusion/optimizer/src/push_down_projection.rs:231]: This change was done in https://github.com/apache/arrow-datafusion/pull/7981 -- it doesn't seem necessary and I think was just a mistake (and did not have adequate test coverage)

**Review by comphead — APPROVED** [review]
lgtm, 
`from`
`from_name`
`from_qualified_name`
might be confusing, we may want to have better naming?

**Comment by alamb** [comment]
> might be confusing, we may want to have better naming?

It is very confusing and results in subtle bugs. I'll see if I can find some way to make it better as a follow on PR

**Comment by alamb** [comment]
Thank you for the quick review @comphead

File: datafusion/optimizer/src/push_down_projection.rs (hunk 1 above)
Comment (gruuya) [reply datafusion/optimizer/src/push_down_projection.rs:231]: Thanks for resolving this; this was a leftover from my initial attempt to address the qualified alias problem (without altering the `Expr::Alias` enum) that I forgot to rollback.

Comment (alamb) [reply datafusion/optimizer/src/push_down_projection.rs:231]: And sadly we had no test coverage for it :(

**Pull request merged**